pyomo/repn/plugins/nl_writer.py

```diff
@@ -1779,7 +1785,8 @@ def _linear_presolve(self, comp_by_linear_var, lcon_by_linear_nnz, var_bounds):
                 a = -coef2 / coef
                 x = id2
                 b = expr_info.const = (lb - expr_info.const) / coef
-                expr_info.linear[x] = a
+                if a != 0:
+                    expr_info.linear[x] = a
```

Comment:
@jsiirola, is this what you meant by "fixing `var_map[_id]`" below? I thought I knew what you meant, but know I'm not so sure. When `a` is `0`, we are replacing variable `_id` with `b`, which, I think, is the same as fixing it. Previously, we were still adding a coefficient of `0` to `expr_info.linear` for `x`, which was unnecessary. I fixed that. Is that what you meant?

Reply:
I was thinking about not having to guard all the bounds manipulation code in the `if a != 0` block.  But, thinking more about it, I don't think that makes much sense: you would need the if `a == 0` block here, and then would still need the guard here (because we still want to finish this block out (to log the debug message and to add the `con_id` to the `eliminated_cons`.